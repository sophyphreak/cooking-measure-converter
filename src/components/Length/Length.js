import React from 'react';
import TableRow from '../../containers/TableRow/TableRow';
import getPropsForTableRow from './getPropsForTableRow/getPropsForTableRow';

const Length = ({ lengthProps }) => {
  const {
    metricDropdown,
    metricInput,
    imperialInput,
    imperialDropdown
  } = getPropsForTableRow(lengthProps);
  return (
    <TableRow
      metricDropdown={metricDropdown}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialDropdown={imperialDropdown}
    />
  );
};

export default Length;
