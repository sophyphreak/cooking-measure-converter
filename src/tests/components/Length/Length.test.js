import React from 'react';
import { shallow } from 'enzyme';
import Length from '../../../components/Length/Length';
import lengthUnitOptions from '../../../containers/MeasureConverter/unitOptions/lengthOptions';

let lengthState,
  onImperialLengthChange,
  onMetricLengthChange,
  lengthProps,
  wrapper;

beforeEach(() => {
  lengthState = {
    imperialLength: '',
    imperialUnit: 'lb',
    metricLength: '',
    metricUnit: 'kg'
  };
  onImperialLengthChange = jest.fn();
  onMetricLengthChange = jest.fn();
  lengthProps = {
    lengthState,
    onImperialLengthChange,
    onMetricLengthChange,
    lengthUnitOptions
  };
  wrapper = shallow(<Length lengthProps={lengthProps} />);
});

test('should render Length correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
