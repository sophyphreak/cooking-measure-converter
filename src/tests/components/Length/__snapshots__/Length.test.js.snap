// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Length correctly 1`] = `
<TableRow
  imperialDropdown={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        "ft",
        "in",
      ],
      "value": "lb",
    }
  }
  imperialInput={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
  metricDropdown={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        "m",
        "cm",
        "mm",
      ],
      "value": "kg",
    }
  }
  metricInput={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
/>
`;
